src/Symfony/Bridge/Monolog/Processor/RouteProcessor.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Monolog\Processor;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\FinishRequestEvent;
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+/**
+ * Adds the current route information to the log entry.
+ *
+ * @author Piotr Stankowski <git@trakos.pl>
+ */
+class RouteProcessor implements ProcessorInterface, EventSubscriberInterface
+{
+    private $routeData = array();
+    private $includeParams;
+
+    public function __construct(bool $includeParams = true)
+    {
+        $this->includeParams = $includeParams;
+    }
+
+    public function __invoke(array $records)
+    {
+        if (!empty($this->routeData) && !isset($records['extra']['requests'])) {
+            $records['extra']['requests'] = $this->routeData;
+        }
+
+        return $records;
+    }
+
+    public function onKernelRequest(GetResponseEvent $event)
+    {
+        if ($event->isMasterRequest()) {
+            $this->routeData = array();
+        }
+
+        $request = $event->getRequest();
+        if (!$request->attributes->has('_controller')) {
+            return;
+        }
+
+        $currentRequestData = array(
+            'controller' => $request->attributes->get('_controller'),
+            'route' => $request->attributes->get('_route'),
+        );
+
+        if ($this->includeParams) {
+            $currentRequestData['route_params'] = $request->attributes->get('_route_params');
+        }
+
+        $this->routeData[] = $currentRequestData;
+    }
+
+    public function onKernelFinishRequest(FinishRequestEvent $event)
+    {
+        if (!$event->getRequest()->attributes->has('_controller')) {
+            return;
+        }
+
+        array_pop($this->routeData);
+    }
+
+    public static function getSubscribedEvents()
+    {
+        return array(
+            KernelEvents::REQUEST => array('onKernelRequest', 1),
+            KernelEvents::FINISH_REQUEST => array('onKernelFinishRequest', 1),
```

Comment:
1. Could you rename theses method to what they do (in opposition to what they listen)
1. Just to know, what did you choose `1` as priority ? (same for the command)

Reply:
1. OK, I've renamed them to addRouteData and removeRouteData
2. Route/command data can be changed by other listeners, and I wanted to get data after all the changes. It could be lower, but I've decided that if something has zero or negative priority, it probably has better reasons